label_studio/projects/functions/next_task.py

```diff
@@ -272,7 +273,8 @@ def get_next_task(user, prepared_tasks, project, dm_queue, assigned_flag=None):
                 count = next_task.annotations.filter(was_cancelled=False).count()
                 task_overlap_reached = count >= next_task.overlap
                 global_overlap_reached = count >= project.maximum_annotations
-                if next_task.is_labeled or task_overlap_reached or global_overlap_reached:
+                locks = next_task.locks.count() > project.maximum_annotations - next_task.annotations.count()
```

Comment:
this whole block introduces some additional queries - let's move it under the feature flag `fflag_fix_back_dev_4185_next_task_additional_logging_long` and enable only on demand